### PR TITLE
✨ Nhj consensus contigs

### PR DESCRIPTION
**Pull request by NHJohnson**
## Description

This gets rid of the rigid canonical version of ALLOWED_CHROMS and adopts instead the contigs provided by the user in a BED file or, if none is provided, the contigs in the Strelka2 VCF header. Variants not in an allowed contig do not appear in the consensus output. 

Fixes https://github.com/d3b-center/bixu-tracker/issues/1085
New Dockerfile in bixtools and update to CWL tool will be needed to actually implement the change in the workflow. 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Tested with and without provided BED file on EC2 instance with inputs from
https://cavatica.sbgenomics.com/u/nathanj/test-project/tasks/d5f469dd-fc9a-4680-bdcd-ca081ea01e04/
Note VEP cache, AF VCF not needed. 
Testing BED file and outputs for both are in 
https://cavatica.sbgenomics.com/u/nathanj/test-project/files/#q?path=consensus_test_output

**Test Configuration**:
* Environment: Python 3.8.5, pysam 0.16.0.1, samtools 1.10
* Test files: See above

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings
- [X] I have committed any related changes to the PR

### CODE REVIEW

**Review by migbro — CHANGES_REQUESTED** [review]
The default mode works well, however I'd like to be able to use an interval bed, and have the script just store the contig names uniquely

File: scripts/consensus_merge.py [hunk 1]
```diff
@@ -527,6 +502,40 @@ def build_output_record(single_caller_variants, output_vcf, sample_names, hotspo
...
+        File must be tab-delimited with contig names in first field
+        Order of contigs in file will be preserved in consensus output
+
+        Args:
+            bed_path (str): path to BED file
+
+        Raises:
+            IOError if bed_path does not point to a file
+                    if no contig names are identified
+                    if contig names are not unique in file
```
[the remark this answers is not in the document]
Comment (NHJohnson) [reply scripts/consensus_merge.py:516]: That would fail in this version. The reason I didn't write it to work the way you had in mind was to preserve the BED order of the contigs in the output. I can change it so that it doesn't require uniqueness and would retain the order of first occurrence of a contig name in the BED file. Would that work for you?

A BED file isn't really necessary since it doesn't use the coordinate information, but if there is a particular such file you had in mind for this purpose, ok.

**Review by dmiller15 — APPROVED** [review]

File: scripts/consensus_merge.py [hunk 2]
```diff
@@ -300,13 +273,14 @@ def write_output_header(output_vcf, sample_list, contig_list, hotspot_source=Non
     output_vcf.header.formats.add('DPR', '1', 'Integer',
             'Difference between highest and lowest DP')
     for contig in sorted(contig_list, key=lambda x: x.id):
-        if strip_chr(contig.name) in ALLOWED_CHROMS:
+        if contig.name in ALLOWED_CHROMS:
```
[the remark this answers is not in the document]
Comment (NHJohnson) [reply scripts/consensus_merge.py:276]: That would be cool. As it is, I could use the Strelka2 header to make a list of 'known contigs' and then raise an error if any provided contig is not on this list, which would at least alert the user to possible errors. As it is, if you miscapitalize a name in the input, it will just silently drop all the calls on the chromosome you thought you were specifying.

File: scripts/consensus_merge.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NHJohnson) [reply scripts/consensus_merge.py:516]: Yes, sounds good. I will also try to incorporate the suggestion attached to Dan's comment about rejecting with an error user-submitted contig names that are not in the Strelka2 header.

File: scripts/consensus_merge.py [hunk 3]
```diff
@@ -527,6 +502,43 @@ def build_output_record(single_caller_variants, output_vcf, sample_names, hotspo
...
+            fields = line.rstrip().split('\t')
+            name = fields[0]
+            if name in contig_names:
+                continue
+            if name not in known_contigs:
+                raise IOError('Contig %s not known from e.g. VCF header' % name)
+            contig_names.append(name)
+
+    if not contig_names:
+        raise IOError('No contig names identified in %s' % bed_path)
```
Comment (dmiller15) [inline scripts/consensus_merge.py:538]: Ok so we're just killing the job if there's a mislabeled contig. Not very permissive but at least the user has a descriptive error for this corner case.

**Review by migbro — APPROVED** [review]
Looks great, tests well locally. Thanks!
